File: app/front/src/components/main/dto/login/jwtTokenDto.ts
interface JwtTokenDtoInterface {

    /** アクセストークン  */
    accessToken: string;

    /** リフレッシュトークン */
    refreshToken: string;

    /** 有効期限 */
    expiresAt: Date;

}

/** Jwt格納Dto */
class JwtTokenDto implements JwtTokenDtoInterface {

    /** アクセストークン  */
    accessToken: string;

    /** リフレッシュトークン */
    refreshToken: string;

    /** 有効期限 */
    expiresAt: Date;

    constructor() {
        const INIT_STRING: string = "";

        this.accessToken = INIT_STRING;
        this.refreshToken = INIT_STRING;
        this.expiresAt = new Date(1948, 7, 29);
    }
}

export { type JwtTokenDtoInterface, JwtTokenDto }

File: app/back/src/main/java/net/seijishikin/jp/normalize/manage/kanrensha/dto/sequrity/JwtTokenDto.java
package net.seijishikin.jp.normalize.manage.kanrensha.dto.sequrity;

import java.io.Serializable;
import java.util.Date;

/**
 * JwtトークンDto
 */
public class JwtTokenDto implements Serializable {

    /** Serialize id */
    private static final long serialVersionUID = 1L;

    /** アクセストークン */
    private final String accessToken;

    /** リフレッシュトークン */
    private final String refreshToken;

    /** 有効期限 */
    private final Date expiresAt;

    /**
     * コンストラクタ
     *
     * @param accessToken  アクセストークン
     * @param refreshToken リフレッシュトークン
     * @param expiresAt    有効期限
     */
    public JwtTokenDto(final String accessToken, final String refreshToken, final Date expiresAt) {
        this.accessToken = accessToken;
        this.refreshToken = refreshToken;
        this.expiresAt = expiresAt;
    }

    /**
     * アクセストークン
     *
     * @return アクセストークン
     */
    public String getAccessToken() {
        return accessToken;
    }

    /**
     * リフレッシュトークン
     *
     * @return リフレッシュトークン
     */
    public String getRefreshToken() {
        return refreshToken;
    }

    /**
     * 有効期限
     *
     * @return 有効期限
     */
    public Date getExpiresAt() {
        return expiresAt;
    }

}
